pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -504,7 +504,6 @@ func (gc *GarbageCollector) monitorFor(resource unversioned.GroupVersionResource
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
 				setObjectTypeMeta(newObj)
```

Comment:
Can you open an issue so we'll revisit whether it's OK for garbage collector to mutate these objects?